src/lib.rs

```diff
@@ -8,6 +8,244 @@ pub mod qdrant;
 #[cfg(feature = "serde")]
 pub mod serde;
 
+use qdrant::{value::Kind::*, ListValue, RetrievedPoint, ScoredPoint, Struct, Value};
+
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+
+static NULL_VALUE: Value = Value {
+    kind: Some(NullValue(0)),
+};
+
+macro_rules! get_payload {
+    ($ty:ty) => {
+        impl $ty {
+            /// get a payload value for the specified key. If the key is not present,
+            /// this will return a null value
+            pub fn get(&self, key: &str) -> &Value {
+                self.payload.get(key).unwrap_or(&NULL_VALUE)
+            }
+        }
+    };
+}
+
+get_payload!(RetrievedPoint);
+get_payload!(ScoredPoint);
+
+macro_rules! extract {
+    ($kind:ident, $check:ident) => {
+        /// check if this value is a
+        #[doc = stringify!($kind)]
+        pub fn $check(&self) -> bool {
+            matches!(self.kind, Some($kind(_)))
+        }
+    };
+    ($kind:ident, $check:ident, $extract:ident, $ty:ty) => {
+        extract!($kind, $check);
+
+        /// extract the contents if this value is a
+        #[doc = stringify!($kind)]
+        pub fn $extract(&self) -> Option<$ty> {
+            if let Some($kind(v)) = self.kind {
+                Some(v)
+            } else {
+                None
+            }
+        }
+    };
+    ($kind:ident, $check:ident, $extract:ident, ref $ty:ty) => {
+        extract!($kind, $check);
+
+        /// extract the contents if this value is a
+        #[doc = stringify!($kind)]
+        pub fn $extract(&self) -> Option<&$ty> {
+            if let Some($kind(v)) = &self.kind {
+                Some(v)
+            } else {
+                None
+            }
+        }
+    };
+}
+
+impl Value {
+    extract!(NullValue, is_null);
+    extract!(BoolValue, is_bool, as_bool, bool);
+    extract!(IntegerValue, is_integer, as_integer, i64);
+    extract!(DoubleValue, is_double, as_double, f64);
+    extract!(StringValue, is_str, as_str, ref String);
+    extract!(ListValue, is_list, as_list, ref [Value]);
+    extract!(StructValue, is_struct, as_struct, ref Struct);
+
+    #[cfg(feature = "serde")]
+    /// convert this into a `serde_json::Value`
+    pub fn into_json(self) -> serde_json::Value {
```

Comment:
I'd love a quick sanity check test validating the round trip `Value` <-> `serde_json::Value`

Reply:
I'm wary of round trips (because errors in the return lap can mask errors in the previous one), so I prefer a simple known state test.